src/wallet/rpcdump.cpp

```diff
@@ -534,9 +534,9 @@ UniValue importwallet(const JSONRPCRequest& request)
         int64_t nFilesize = std::max((int64_t)1, (int64_t)file.tellg());
         file.seekg(0, file.beg);
 
-        pwallet->ShowProgress(_("Importing..."), 0); // show progress dialog in GUI
+        uiInterface.ShowProgress(_("Importing..."), 0, false); // show progress dialog in GUI
```

Comment:
In commit "Add cancel button to rescan progress dialog"

These changes from wallet to global `ShowProgress` calls seem unrelated. Are they intentional? It might make sense to change these if removing the wallet `ShowProgress` signal, but since that is still used in `CWallet::ScanForWalletTransactions`, this change just makes these calls inconsistent.

If this change is necessary it would be good to have a comment saying why CWallet::ShowProgress is not used here.

Reply:
Because `CWallet::ShowProgress` has a cancel button now which is directly tied to `AbortRescan` (because it isn't used for anything except showing rescan progress), I needed a different progress bar which does not have a cancel button. This change makes that `ShowProgress` use `BitcoinGUI::ShowProgress` (which is tied toe `uiInterface`)which does not have a cancel button (and is also not used anywhere else AFAICT). This progress bar indicates the import progress, not rescan progress which is why a different `ShowProgress` is needed